diff -urN ilmbase-1.0.1/Half/half.h ilmbase/Half/half.h
--- ilmbase-1.0.1/Half/half.h	2007-05-03 10:22:26.000000000 +0800
+++ ilmbase/Half/half.h	2009-08-16 18:19:14.000000000 +0800
@@ -87,7 +87,9 @@
 
 #include <iostream>
 
-class half
+#include <IlmBaseConfig.h>
+
+class DLL_API half
 {
   public:
 
diff -urN ilmbase-1.0.1/Iex/IexBaseExc.h ilmbase/Iex/IexBaseExc.h
--- ilmbase-1.0.1/Iex/IexBaseExc.h	2006-12-09 05:59:36.000000000 +0800
+++ ilmbase/Iex/IexBaseExc.h	2009-08-16 17:35:10.000000000 +0800
@@ -49,6 +49,8 @@
 #include <exception>
 #include <sstream>
 
+#include <IlmBaseConfig.h>
+
 namespace Iex {
 
 #if (defined _WIN32 || defined _WIN64) && defined _MSC_VER
@@ -60,7 +62,7 @@
 // Our most basic exception class
 //-------------------------------
 
-class BaseExc: public std::string, public std::exception
+class DLL_API BaseExc: public std::string, public std::exception
 {
   public:
 
@@ -126,7 +128,7 @@
 //-----------------------------------------------------
 
 #define DEFINE_EXC(name, base)				        \
-    class name: public base				        \
+    class DLL_API name: public base				        \
     {							        \
       public:                                                   \
 	name (const char* text=0)      throw(): base (text) {}	\
diff -urN ilmbase-1.0.1/Iex/IexThrowErrnoExc.h ilmbase/Iex/IexThrowErrnoExc.h
--- ilmbase-1.0.1/Iex/IexThrowErrnoExc.h	2006-12-09 05:59:36.000000000 +0800
+++ ilmbase/Iex/IexThrowErrnoExc.h	2009-08-16 20:03:48.000000000 +0800
@@ -87,8 +87,8 @@
 //
 //--------------------------------------------------------------------------
 
-void throwErrnoExc (const std::string &txt, int errnum);
-void throwErrnoExc (const std::string &txt = "%T." /*, int errnum = oserror() */);
+DLL_API void throwErrnoExc (const std::string &txt, int errnum);
+DLL_API void throwErrnoExc (const std::string &txt = "%T." /*, int errnum = oserror() */);
 
 
 } // namespace Iex
diff -urN ilmbase-1.0.1/IlmThread/IlmThreadMutex.h ilmbase/IlmThread/IlmThreadMutex.h
--- ilmbase-1.0.1/IlmThread/IlmThreadMutex.h	2006-12-12 12:17:35.000000000 +0800
+++ ilmbase/IlmThread/IlmThreadMutex.h	2009-08-16 20:13:06.000000000 +0800
@@ -83,7 +83,7 @@
 class Lock;
 
 
-class Mutex
+class DLL_API Mutex
 {
   public:
 
@@ -108,7 +108,7 @@
 };
 
 
-class Lock
+class DLL_API Lock
 {
   public:
 
diff -urN ilmbase-1.0.1/IlmThread/IlmThreadPool.h ilmbase/IlmThread/IlmThreadPool.h
--- ilmbase-1.0.1/IlmThread/IlmThreadPool.h	2006-12-09 05:59:36.000000000 +0800
+++ ilmbase/IlmThread/IlmThreadPool.h	2009-08-16 20:27:00.000000000 +0800
@@ -60,13 +60,15 @@
 //
 //-----------------------------------------------------------------------------
 
+#include <IlmBaseConfig.h>
+
 namespace IlmThread {
 
 class TaskGroup;
 class Task;
 
 
-class ThreadPool  
+class DLL_API ThreadPool  
 {
   public:
 
@@ -123,7 +125,7 @@
 };
 
 
-class Task
+class DLL_API Task
 {
   public:
 
@@ -139,7 +141,7 @@
 };
 
 
-class TaskGroup
+class DLL_API TaskGroup
 {
   public:
 
diff -urN ilmbase-1.0.1/IlmThread/IlmThreadSemaphore.h ilmbase/IlmThread/IlmThreadSemaphore.h
--- ilmbase-1.0.1/IlmThread/IlmThreadSemaphore.h	2006-12-12 12:17:47.000000000 +0800
+++ ilmbase/IlmThread/IlmThreadSemaphore.h	2009-08-16 20:13:34.000000000 +0800
@@ -59,7 +59,7 @@
 namespace IlmThread {
 
 
-class Semaphore
+class DLL_API Semaphore
 {
   public:
 
diff -urN ilmbase-1.0.1/Imath/ImathRandom.h ilmbase/Imath/ImathRandom.h
--- ilmbase-1.0.1/Imath/ImathRandom.h	2007-03-15 08:23:17.000000000 +0800
+++ ilmbase/Imath/ImathRandom.h	2009-08-16 18:02:00.000000000 +0800
@@ -59,6 +59,8 @@
 #include <stdlib.h>
 #include <math.h>
 
+#include <IlmBaseConfig.h>
+
 namespace Imath {
 
 //-----------------------------------------------
@@ -67,7 +69,7 @@
 // length of 2^32.
 //-----------------------------------------------
 
-class Rand32
+class DLL_API Rand32
 {
   public:
 
@@ -127,7 +129,7 @@
 // uniformly distributed sequence.
 //--------------------------------------------------------
 
-class Rand48
+class DLL_API Rand48
 {
   public:
 
diff -urN ilmbase-1.0.1/config.windows/IlmBaseConfig.h ilmbase/config.windows/IlmBaseConfig.h
--- ilmbase-1.0.1/config.windows/IlmBaseConfig.h	2006-12-13 06:27:40.000000000 +0800
+++ ilmbase/config.windows/IlmBaseConfig.h	2009-08-16 20:32:16.000000000 +0800
@@ -14,3 +14,10 @@
 
 #undef HAVE_POSIX_SEMAPHORES
 
+#if defined(OPENEXR_BUILD_DLL)
+	#define DLL_API __declspec(dllexport)
+#elif defined(OPENEXR_LINK_DLL)
+	#define DLL_API __declspec(dllimport)
+#else
+	#define DLL_API
+#endif
